CMakeLists.txt

```diff
@@ -330,16 +330,27 @@ add_library(wabt STATIC ${WABT_LIBRARY_SRC})
 
 # libwasm, which implenents the wasm C API
 if (BUILD_LIBWASM)
-  add_library(wasm SHARED ${WABT_LIBRARY_SRC} src/interp/interp-wasm-c-api.cc)
-  target_link_libraries(wasm wabt)
-  target_include_directories(wasm PUBLIC third_party/wasm-c-api/include)
-  if (COMPILER_IS_MSVC)
-    target_compile_definitions(wasm PRIVATE "WASM_API_EXTERN=__declspec(dllexport)")
-  else ()
-    target_compile_options(wasm PRIVATE -Wno-old-style-cast)
-    target_compile_definitions(wasm PRIVATE "WASM_API_EXTERN=__attribute__((visibility(\"default\")))")
-  endif ()
-  set_target_properties(wasm PROPERTIES CXX_VISIBILITY_PRESET hidden)
+  foreach (LIBTYPE SHARED STATIC)
+    if (LIBTYPE STREQUAL "SHARED")
+      set(LIBNAME "wasm")
+    else()
+      set(LIBNAME "wasm_static")
+    endif()
+    add_library(${LIBNAME} ${LIBTYPE} src/interp/interp-wasm-c-api.cc)
+    # TODO: add just the minimum files needed for the interp
+    # to minimize size of the static library?
+    target_link_libraries(${LIBNAME} wabt)
+    target_include_directories(${LIBNAME} PUBLIC third_party/wasm-c-api/include)
+    if (COMPILER_IS_MSVC)
+      target_compile_definitions(${LIBNAME} PRIVATE "WASM_API_EXTERN=__declspec(dllexport)")
+    else ()
+      target_compile_options(${LIBNAME} PRIVATE -Wno-old-style-cast)
+      target_compile_definitions(${LIBNAME} PRIVATE "WASM_API_EXTERN=__attribute__((visibility(\"default\")))")
+    endif ()
+    if (LIBTYPE STREQUAL "SHARED")
```

Comment:
It was necessary when building locally, but has dropped out of my brain's L1/L2 cache -- I can dig up the reason if you like, but it's moot since I'm withdrawing this PR (see below)